presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -467,7 +462,7 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
                     .orElse(icebergStatistics);
             TableStatistics.Builder filteredStatsBuilder = TableStatistics.builder()
                     .setRowCount(mergedStatistics.getRowCount());
-            for (ColumnHandle colHandle : columnHandles) {
+            for (ColumnHandle colHandle : icebergStatistics.getColumnStatistics().keySet()) {
```

Comment:
nit: Since there are multiple usages of `icebergStatistics.getColumnStatistics().keySet()` we can probably assign it to a variable.